### PR TITLE
Migrating 'Assigning PCI devices to VM'

### PR DESCRIPTION
**Pull request by tbazant**
### Is this (based on) existing content?

* old xml:id `task-assign-pci-device-libvirt` -> `vm-assign-pci-device`
* old URL: https://documentation.suse.com/smart/virtualization-cloud/html/task-assign-pci-device-libvirt/task-assign-pci-device-libvirt.html

### CODE REVIEW

File: tasks/vm-assign-pci-device-virsh.xml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        </para>
+      </tip>
+      <para>
+        In the example above, the <literal>managed='yes'</literal> option means that the device is
+        managed. To switch the device mode to unmanaged, set <literal>managed='no'</literal>. If
+        you do so, you need to take care of the related driver with the <command>virsh
+        nodedev-detach</command> and <command>virsh nodedev-reattach</command> commands. Before
+        starting the &vmguest; you need to detach the device from the host by running
+      </para>
+<screen>&prompt.sudo;virsh nodedev-detach <replaceable>pci_0000_03_07_0</replaceable></screen>
```
Comment (lvicoun) [inline tasks/vm-assign-pci-device-virsh.xml:157]: This command is already mentioned as step 3.

File: tasks/vm-assign-pci-device-virsh.xml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+          driver.
+        </para>
+        <para>
+          If the device is unmanaged, you must manually manage these tasks before assigning the
+          device to a domain, and after the device is no longer used by the domain.
+        </para>
+      </tip>
+      <para>
+        In the example above, the <literal>managed='yes'</literal> option means that the device is
+        managed. To switch the device mode to unmanaged, set <literal>managed='no'</literal>. If
```
Comment (lvicoun) [inline tasks/vm-assign-pci-device-virsh.xml:152]: I'm not sure if the intention is to set to unmanaged in this case. 
```suggestion
        managed. If the device mode is unmanaged, the option <literal>managed='yes'</literal> value is <literal>'yes'</literal>, 
```

File: tasks/vm-assign-pci-device-virsh.xml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        you do so, you need to take care of the related driver with the <command>virsh
```
Comment (lvicoun) [inline tasks/vm-assign-pci-device-virsh.xml:153]:
```suggestion
        you need to take care of the related driver with the <command>virsh
```

File: tasks/vm-assign-pci-device-virsh.xml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        nodedev-detach</command> and <command>virsh nodedev-reattach</command> commands. Before
```
Comment (lvicoun) [inline tasks/vm-assign-pci-device-virsh.xml:154]: I'd suggest to provide substeps in this case.

File: tasks/vm-assign-pci-device-virsh.xml [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      <para>
+        Run <command>virsh edit</command> on your domain, and add the following device entry in the
+        <literal>&lt;devices&gt;</literal> section using the result from the previous step:
+      </para>
+<screen>&lt;hostdev mode='subsystem' type='pci' managed='yes'&gt;
+   &lt;source&gt;
+     &lt;address domain='0x0' bus='0x03' slot='0x07' function='0x0'/&gt;
+   &lt;/source&gt;
+ &lt;/hostdev&gt;</screen>
+      <tip xml:id="tip-libvirt-config-pci-virsh-managed">
```
Comment (lvicoun) [inline tasks/vm-assign-pci-device-virsh.xml:130]: I'd move this into a separate section, here it breaks the procedure flow.

File: tasks/vm-assign-pci-device-virsh.xml [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+      </tip>
+      <para>
+        In the example above, the <literal>managed='yes'</literal> option means that the device is
+        managed. To switch the device mode to unmanaged, set <literal>managed='no'</literal>. If
+        you do so, you need to take care of the related driver with the <command>virsh
+        nodedev-detach</command> and <command>virsh nodedev-reattach</command> commands. Before
+        starting the &vmguest; you need to detach the device from the host by running
+      </para>
+<screen>&prompt.sudo;virsh nodedev-detach <replaceable>pci_0000_03_07_0</replaceable></screen>
+      <para>
```
Comment (lvicoun) [inline tasks/vm-assign-pci-device-virsh.xml:158]: Please consider this as a note/tip how to use the device on host again. As according to the flow, the VM guest is now running, so you should not be able to attach it to the host machine.
If this is command is the only one to make it available again, please move it outside of the procedure as it is not related to assigning the device to the VM guest machine.

File: tasks/vm-assign-pci-device-vmm.xml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+<!DOCTYPE topic
+[
+  <!ENTITY % entities SYSTEM "../common/generic-entities.ent">
+    %entities;
+]>
+<!-- refers to legacy doc: <add github link to legacy doc piece, if applicable> -->
+<!-- point back to this document with a similar comment added to your legacy doc piece -->
+<!-- refer to README.md for file and id naming conventions -->
+<!-- metadata is dealt with on the assembly level -->
+<topic xml:id="vm-assign-pci-device-vmm"
```
Comment (lvicoun) [inline tasks/vm-assign-pci-device-vmm.xml:13]: What about SELinux? No changes required if attaching the device in VManager?
The PCI device handling (managed vs unamanaged) is not related to this type of PCI device assignment?

**Review by lvicoun — CHANGES_REQUESTED** [review]

File: tasks/vm-assign-pci-device-vmm.xml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+ xmlns:trans="http://docbook.org/ns/transclusion">
+  <info>
+    <title>Assigning a PCI device to a virtual machine using &vmm;</title>
+    <meta name="maintainer" content="tbazant@suse.com" its:translate="no"/>
+    <abstract>
+      <para>
+        Assigning a &vmhost;'s PCI device to a &vmguest;&mdash;also known as
+        &pciback;&mdash;dedicates the PCI device's resources to the &vmguest;. After the PCI device
+        is assigned, it cannot be used by the &vmhost; or by any other &vmguest; on the same
+        &vmhost;. The following procedure describes how to assign &vmhost;'s PCI device to a
```
Comment (dariavladykina) [inline tasks/vm-assign-pci-device-vmm.xml:28]:
```suggestion
        &vmhost;. The following procedure describes how to assign a &vmhost;'s PCI device to a
```

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some minor suggestions. Thanks!

**Review by lvicoun — CHANGES_REQUESTED** [review]
Thanks Tomas! Please, see my comments and apply what you consider valuable :)

File: tasks/vm-assign-pci-device-unmanaged.xml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        &libvirt; is a unified interface to multiple virtualization solutions. &libvirt; virtual
+        machines are typically managed by either &virsh;, a command-line utility, or &vmm;, an
+        application with a graphical user interface. Assigning a &vmhost;'s PCI (or USB) devices to
+        a &vmguest;&mdash;also known as &pciback;&mdash;dedicates the device's resources to the
+        &vmguest;.
+      </para>
+    </abstract>
+  </info>
+  <para>
+    When assigning a host device to a &vmguest;, you can specify whether the device is either
```
Comment (dariavladykina) [inline tasks/vm-assign-pci-device-unmanaged.xml:34]:
```suggestion
    When assigning a host device to a &vmguest;, you can specify whether the device is
```

File: tasks/vm-assign-pci-device-unmanaged.xml [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+  </para>
+<screen>&lt;hostdev mode='subsystem' type='pci' managed='yes'&gt;</screen>
+  <para>
+    <emphasis>Managed</emphasis> devices are automatically detached from the host driver and
+    attached to the virtual <systemitem>vfio-pci</systemitem> driver of the guest during the domain
+    startup. When the domain is terminated, the device is reattached to the original host driver.
+  </para>
+  <para>
+    For <emphasis>unmanaged devices</emphasis>, you need to take care of detaching the device from
+    the host and attaching it to the <systemitem>vfio-pci</systemitem> driver manually as suggested
```
Comment (dariavladykina) [inline tasks/vm-assign-pci-device-unmanaged.xml:47]:
```suggestion
    the host and attaching it to the <systemitem>vfio-pci</systemitem> driver manually, as suggested
```

File: tasks/vm-assign-pci-device-vmm.xml [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+          </imageobject>
+          <imageobject role="html">
+            <imagedata fileref="virt_add_pcidevice.png" width="75%"/>
+          </imageobject>
+        </mediaobject>
+      </figure>
+    </step>
+    <step>
+      <para>
+        From the list of available PCI devices, choose the one you want to pass to the guest.
```
Comment (dariavladykina) [inline tasks/vm-assign-pci-device-vmm.xml:68]:
```suggestion
        From the list of PCI devices, choose the one you want to pass to the guest.
```

File: tasks/vm-assign-pci-device-vmm.xml [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+      </para>
+    </step>
+    <step>
+      <para>
+        Click <guimenu>Add Hardware</guimenu> and choose the <guimenu>PCI Host Device</guimenu>
+        category in the left panel. A list of available PCI devices appears in the right part of
+        the window.
+      </para>
+      <figure>
+        <title>Adding a PCI device</title>
```
Comment (dariavladykina) [inline tasks/vm-assign-pci-device-vmm.xml:55]: Please add alt text to the figure with `<textobject role="description"/>`.

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some comments here. Thanks!

File: tasks/vm-assign-pci-device-virsh.xml [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+ xmlns="http://docbook.org/ns/docbook" version="5.2"
+ xmlns:its="http://www.w3.org/2005/11/its"
+ xmlns:xi="http://www.w3.org/2001/XInclude"
+ xmlns:xlink="http://www.w3.org/1999/xlink"
+ xmlns:trans="http://docbook.org/ns/transclusion">
+  <info>
+    <title>Assigning a device to a virtual machine using &virsh;</title>
+    <meta name="maintainer" content="tbazant@suse.com" its:translate="no"/>
+    <abstract>
+      <para>
```
Comment (janajaeger) [inline tasks/vm-assign-pci-device-virsh.xml:24]: This is not specific to the the task being done with virsh, but rather to the process of adding hw, right? If so, it doesn't belong here, but one level up in the intro, I'd say.

File: tasks/vm-assign-pci-device-vmm.xml
Comment (janajaeger) [inline tasks/vm-assign-pci-device-vmm.xml:1]: I assume you have coordinated with Shalaka? IIRC, she was or is still working on a virt-manager piece?

**Review by janajaeger — COMMENTED** [review]
See the files for a few nitpicks. Mainly about missing use cases and troubleshooting. Nice read, though ;) Thanks a million for the migration!

**Pull request merged**